[BUG] - Azure deployment failing because of Unsupported argument in the AKS cluster Terraform configuration

### Describe the bug

Azure deployments are now failing with the following error:
```
Error:  2025-04-07T14:55:05.155Z [ERROR] vertex "module.kubernetes.azurerm_kubernetes_cluster.main" error: Unsupported argument
[tofu]: ╷
[tofu]: │ Error: Unsupported argument
[tofu]: │ 
[tofu]: │   on modules/kubernetes/main.tf line 40, in resource "azurerm_kubernetes_cluster" "main":
[tofu]: │   40:     enable_auto_scaling = "true"
[tofu]: │ 
[tofu]: │ An argument named "enable_auto_scaling" is not expected here.
[tofu]: ╵
```

### Expected behavior

Deployment should succeed.

### OS and architecture in which you are running Nebari

Not relevant

### How to Reproduce the problem?

Deploy Nebari to Azure off https://github.com/nebari-dev/nebari/commit/35e32545d93bab03dcd76bee0f3b9d09fd5632e0

### Command output

```bash session

```

### Versions and dependencies used.

_No response_

### Compute environment

None

### Integrations

_No response_

### Anything else?

@viniciusdc found the culprit here: https://github.com/nebari-dev/nebari/pull/2605/files#diff-6f5cccd4b6b298fb389476a02d7596dcd7a9a1f376609e5d366052facef1c7aa

--- a/src/_nebari/stages/infrastructure/template/azure/modules/kubernetes/main.tf
+++ b/src/_nebari/stages/infrastructure/template/azure/modules/kubernetes/main.tf
@@ -34,13 +34,13 @@ resource "azurerm_kubernetes_cluster" "main" {
 
   kubernetes_version = var.kubernetes_version
   default_node_pool {
-    vnet_subnet_id      = var.vnet_subnet_id
-    name                = var.node_groups[0].name
-    vm_size             = var.node_groups[0].instance_type
-    enable_auto_scaling = "true"
-    min_count           = var.node_groups[0].min_size
-    max_count           = var.node_groups[0].max_size
-    max_pods            = var.max_pods
+    vnet_subnet_id       = var.vnet_subnet_id
+    name                 = var.node_groups[0].name
+    vm_size              = var.node_groups[0].instance_type
+    auto_scaling_enabled = "true"
+    min_count            = var.node_groups[0].min_size
+    max_count            = var.node_groups[0].max_size
+    max_pods             = var.max_pods
     # It's not possible to add node_taints to the default node pool. See https://github.com/hashicorp/terraform-provider-azurerm/issues/9183 for more info
 
     orchestrator_version = var.kubernetes_version

